pkg/git/git.go

```diff
@@ -0,0 +1,58 @@
+// Copyright The Shipwright Contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package git
+
+import (
+	"fmt"
+
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/go-git/go-git/v5/storage/memory"
+
+	gogitv5 "github.com/go-git/go-git/v5"
+)
+
+const (
+	defaultRemote = "origin"
+	httpsProtocol = "https"
+	httpProtocol  = "http"
+	fileProtocol  = "file"
+	gitProtocol   = "ssh"
+)
+
+// ValidateGitURLExists validate if a source URL exists or not
+// Note: We have an upcoming PR for the Build Status, where we
+// intend to define a single Status.Reason in the form of 'remoteRepositoryUnreachable',
+// where the Status.Message will contain the longer text, like 'invalid source url
+func ValidateGitURLExists(urlPath string) error {
+
+	endpoint, err := transport.NewEndpoint(urlPath)
+	if err != nil {
+		return err
+	}
+
+	switch endpoint.Protocol {
+	case httpsProtocol, httpProtocol:
```

Comment:
I think @bigkevmcd 's suggestion at https://github.com/shipwright-io/build/pull/512#discussion_r559672019 is a good idea 

So have a `case httpProtocol:` that prints that message, and then fallsthrough  to the `httpsProtocol` block.

Othwerise I'm good @qu1queee

Reply:
@bigkevmcd this can only be logged out if the validation fails. Just to be clear, you say that if an `http` protocol is used, we should enhance `remote repository unreachable` to be something like `insecurely accessing the repository failed, repository unreachable` ?

Reply:
I think any time we parse an `httpProtocol` case, we should log this out as a possibly insecure case.

Reply:
@bigkevmcd thanks. I see. Do you mind if I take this enhancement into the other PR we are gonna make available soon for review, which is #536 . At the moment Build have only a `Status.Reason` which is a message, what we are gonna do is to make `Status.Reason` a one-word camelCase and introduce `Status.Message`, see [field changes](https://github.com/shipwright-io/build/pull/536/files#diff-ff8d190dc59bafe236a40c0df3f74e9528e0441d0493c588d4dab0e5c4ac62d0R254-R260) .With `Status.Message`, we can then embed there the `as a possibly insecure case` whenever we validate an `httpProtocol`.

Reply:
I like the idea of incorporating this into the #536 work @qu1queee - good idea 

That said, I don't see why that would preclude us from still providing a simple informational log here.  Catch the user's attention both in real time (via the log) and post-mortem (via the Build status).

That would be my preference.